scripts/sensorshow

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/python3
+
+"""
+    Script to show Voltage and Current Sensor status.
+"""
+from tabulate import tabulate
+from swsscommon.swsscommon import SonicV2Connector
+from natsort import natsorted
+import argparse
+
+
+header = ['Sensor', 'Voltage(mV)', 'High TH', 'Low TH', 'Crit High TH', 'Crit Low TH', 'Warning', 'Timestamp']
+
+TIMESTAMP_FIELD_NAME = 'timestamp'
+HIGH_THRESH_FIELD_NAME = 'high_threshold'
+LOW_THRESH_FIELD_NAME = 'low_threshold'
+CRIT_HIGH_THRESH_FIELD_NAME = 'critical_high_threshold'
+CRIT_LOW_THRESH_FIELD_NAME = 'critical_low_threshold'
+WARNING_STATUS_FIELD_NAME = 'warning_status'
+
+
+class SensorShow(object):
+    def __init__(self, type):
+        self.db = SonicV2Connector(host="127.0.0.1")
+        self.db.connect(self.db.STATE_DB)
+        self.field_name = type
+        if type == "voltage":
+            self.table_name = 'VOLTAGE_INFO'
+            header[1]  = 'Voltage(mV)'
+        else:
+            self.table_name = 'CURRENT_INFO'
+            header[1]  = 'Current(mA)'
+
+    def show(self):
+        keys = self.db.keys(self.db.STATE_DB, self.table_name + '*')
+        if not keys:
+            print('Sensor Not detected\n')
+            return
+
+        table = []
+        for key in natsorted(keys):
+            key_list = key.split('|')
+            if len(key_list) != 2: # error data in DB, log it and ignore
+                print('Warn: Invalid key in table {}: {}'.format(self.table_name, key))
+                continue
+
+            name = key_list[1]
+            data_dict = self.db.get_all(self.db.STATE_DB, key)
+            table.append((name, 
+                          data_dict[self.field_name], 
+                          data_dict[HIGH_THRESH_FIELD_NAME],
+                          data_dict[LOW_THRESH_FIELD_NAME],
+                          data_dict[CRIT_HIGH_THRESH_FIELD_NAME],
+                          data_dict[CRIT_LOW_THRESH_FIELD_NAME],
+                          data_dict[WARNING_STATUS_FIELD_NAME],
+                          data_dict[TIMESTAMP_FIELD_NAME]
+                          ))
+        
+        if table:
+            print(tabulate(table, header, tablefmt='simple', stralign='right'))
+        else:
+            print('No sensor data available\n')
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-t", "--type", help="sensor type", required=True, default="voltage", choices=['voltage', 'current'])
```

Comment:
I think it would be best to require the sensor type argument, and not default to anything specific, since this may scale up further.

Reply:
ok